pytorch_lightning/cluster_environments/lsf_environment.py

```diff
@@ -0,0 +1,71 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+from pytorch_lightning import _logger as log
+from pytorch_lightning.cluster_environments.cluster_environment import ClusterEnvironment
+
+
+class LSFEnvironment(ClusterEnvironment):
+
+    def __init__(self):
+        super().__init__()
+
+    def master_address(self):
+        # figure out the root node addr
+        try:
+            root_node = sorted(set(x for x in open(os.environ['LSB_DJOB_HOSTFILE'], 'r')
+                                   if 'batch' not in x and 'login' not in x))[0][:-1]
+        except Exception:
+            root_node = "127.0.0.1"
```

Comment:
Can you change it to not use a catch-all exception? maybe something like

```python
with open(os.environ[...]) as f:
   lines = [l.rstrip() for l in f]
   nodes = {l for l in lines if "batch" not in l and "login" not in l}
   if not nodes:
       # give an appropriate warning
   root_node = sorted(nodes)[0]
   if not root_node:
       # give an appropriate warning
       root_node = "127.0.0.1"
```
you can use `.rstrip()` to remove the line breaks after each line instead of relying on `[:-1]` (if that is what you meant to do)